src/main/java/net/clgd/ccemux/emulation/CCEmuXAPI.java

```diff
@@ -107,6 +110,58 @@ public CCEmuXAPI(CCEmuX emu, EmulatedComputer computer, String name) {
 
 			return new Object[] {};
 		});
+
+		methods.put("attach", o -> {
+			if (o.length > 1 && o[0] instanceof String && o[1] instanceof String) {
+				String sideName = (String) o[0];
+				String peripheralName = (String) o[1];
+				int sideId = parseSide(sideName);
+				IPeripheral peripheral = PeripheralFactory.implementations.get(peripheralName).create();
+
+				if(sideId > 0) {
+					if(peripheral != null) {
+						computer.setPeripheral(sideId, peripheral);
+					} else {
+						throw new LuaException("invalid peripheral");
+					}
+				} else {
+					throw new LuaException("invalid side");
+				}
+			} else if (o.length > 1 && !(o[1] instanceof String)) {
+				throw new LuaException("expected string for argument #2");
+			} else {
+				throw new LuaException("expected string for argument #1");
+			}
+
+			return new Object[] {};
+		});
+
+		methods.put("detach", o -> {
+			if (o.length > 0 && o[0] instanceof String) {
+				String sideName = (String) o[0];
+				int sideId = parseSide(sideName);
+
+				if (sideId > 0) {
+					computer.setPeripheral(sideId, null);
+				} else {
+					throw new LuaException("invalid side");
+				}
+			} else {
+				throw new LuaException("expected string for argument #1");
+			}
+
+			return new Object[] {};
+		});
+	}
+
+	private int parseSide(String side) {
+		for(int n = 0; n < Computer.s_sideNames.length; ++n) {
+			if(side.equals(Computer.s_sideNames[n])) {
+				return n;
```

Comment:
Yeah, but when you're doing that, you're creating an entire list just to call `indexOf` on it. Since this isn't performance critical code, I guess it's fine, but if we have `ArrayUtils`, we might as well use that.